keyboards/zlabkeeb/6pad/keymaps/via/keymap.c

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2024 zlabkeeb (zlabkeeb@gmail.com)
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+
+uint16_t underglow_timer = 0;
+bool underglow_on = true;
+uint8_t prev_rgb_mode; // variable to store the previous RGB mode
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    prev_rgb_mode = rgblight_get_mode(); // store the current RGB mode before changing it
```

Comment:
Not sure about the value of the store/revert when nothing is setting it.

Also <https://docs.qmk.fm/#/feature_rgblight?id=lighting-layers> would be less invasive.

Reply:
I will delete it